src/ProtectedProxies/ProtectedUUPSUpgradeable.sol

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: UNLICENSED
+// (c) SphereX 2023 Terms&Conditions
+
+pragma solidity ^0.8.0;
+
+import {UUPSUpgradeable} from "openzeppelin/proxy/utils/UUPSUpgradeable.sol";
+import {Address} from "openzeppelin/utils/Address.sol";
+import {StorageSlot} from "openzeppelin/utils/StorageSlot.sol";
+
+/**
+ * @dev UUPSUpgradeable implementation designed for implementations under SphereX's ProtectedERC1967SubProxy
```

Comment:
explain to me the porpuse of this contract if the ProtectedERC1967SubProxy inherit from the SphereXProtectedSubProxy that implements upgrae mechanism

Reply:
I think it is good enough that they are internal in the proxy, it is how it works in the original ERC1967Proxy which also has an internal upgrade functions.